org-alert.el

```diff
@@ -65,14 +68,10 @@ If nil, never stop sending notifications."
   :group 'org-alert
   :type 'string)
 
-(defcustom org-alert-match-string
-  "SCHEDULED>=\"<today>\"+SCHEDULED<\"<tomorrow>\"|DEADLINE>=\"<today>\"+DEADLINE<\"<tomorrow>\""
-  "property/todo/tags match string to be passed to `org-map-entries'."
-  :group 'org-alert
-  :type 'regexp)
-
 (defcustom org-alert-time-match-string
-  "\\(?:SCHEDULED\\|DEADLINE\\):.*<.*\\([0-9]\\{2\\}:[0-9]\\{2\\}\\).*>"
+  (rx "<" (*? nonl)
+	  (group (= 2 (any "0-9")) ":" (= 2 (any "0-9")))
+	  (* nonl) (or ">" "-"))
```

Comment:
I also didn't know about `rx`. While it's pretty neat, I would slightly prefer using the normal string syntax. At least it seems like evaluating this won't show it to the user. When I evaluate it, I get 

```
"<.*?\\([0-9]\\{2\\}:[0-9]\\{2\\}\\).*[>-]"
```

which is almost the same but not quite as the one in the current code:

```
<.*\\([0-9]\\{2\\}:[0-9]\\{2\\}\\).*>"
```

Is there a reason for the new `?` (non-greedy?) and the trailing `-`?

Reply:
This is, again, tied to my use case.
I use [org-gcal](https://github.com/kidd/org-gcal.el) to sync my events, and I mostly got the `<2024-09-27 Fri 10:00-11:00>` format.

But I think it's reasonable default, since it should handle the default format, and it makes more sense to alert using the earlier time when we actually schedule timestamps with range formats.

Now that I think more about it, the regex should consume the rest without needing the trailing `-`, so I'll remove it. But still, I think the non-greedy syntax is a better default.

<hr>

Regarding `rx`: I'm so used to read the `rx` format instead of regex, to the point I actually have downloaded `xr` package to revert string regex back to `rx` format. Its just my preference leaking into others package 😂.

I'll change this back to the string syntax, and let me know if you want the non-greedy changes on a separate PR instead. I'll happy to do so!